discovery/syncer_test.go

```diff
@@ -335,6 +338,12 @@ func TestGossipSyncerFilterGossipMsgsAllInMemory(t *testing.T) {
 				"messages: %v", len(msgs), spew.Sdump(msgs))
 		}
 	}
+
+	// Wait for error from goroutine.
+	err := <-errCh
```

Comment:
needs to select on a timeout channel as well, in case nothing ever arrives on the `errCh`

Reply:
ok